include/ck/utility/amd_gemm_dpp.hpp

```diff
@@ -5,17 +5,39 @@
 
 #include "ck/utility/common_header.hpp"
 #include "ck/utility/math.hpp"
-#include "ck/utility/amd_gemm_dpp.hpp"
+#include "ck/utility/inner_product_dpp8.hpp"
 
 namespace ck {
 
 namespace dpp8 {
 
-/// Number of lanes that can share data using DPP8 modifiers.
-constexpr index_t lane_group_size = 8;
-
-__device__ index_t get_lane_group_local_idx() { return threadIdx.x / lane_group_size; }
-__device__ index_t get_thread_idx_in_lane_group() { return threadIdx.x % lane_group_size; }
+template <index_t MPerLanegroup,
+          index_t NPerLanegroup,
+          index_t KPerLanegroup,
+          class FloatA,
+          class FloatB,
+          class FloatVecC,
+          bool ShareA>
+__device__ void RunGemm(const FloatA& a, const FloatB& b, FloatVecC& c_vec)
+{
+    constexpr index_t c_dim = ShareA ? MPerLanegroup : NPerLanegroup;
+
+    const vector_type<half_t, KPerLanegroup> a_vector{a};
+    const vector_type<half_t, KPerLanegroup> b_vector{b};
+
+    static_for<0, c_dim, 1>{}([&](auto c_idx) {
+        float c = c_vec.template AsType<float>()(c_idx);
+        // Next `c_idx` implies that we need to pull data from the next lane.
+        constexpr index_t source_lane = c_idx;
+        static_for<0, KPerLanegroup / 2, 1>{}([&](auto k_chunk) {
+            const auto a_half2 = a_vector.template AsType<half2_t>()[k_chunk];
+            const auto b_half2 = b_vector.template AsType<half2_t>()[k_chunk];
+            ck::dpp8::inner_product_dpp<half2_t, half2_t, float, source_lane, ShareA>(
```

Comment:
Why are you using this `half2_t` data type ? If there's any correlation between input data type and dpp instruction data type, then maybe you need some kind of simple type trait structures to get appropriate type? Or if it's very simple rule like just get vector of 2, then you can use `vector_type_maker`

Reply:
I introduced a `dpp_datatypes` struct and changed these parts a bit in general. Now I do not use hardcoded data types